### ISSUE TITLE
Required form fields must contain a required attribute for accessibility

### ISSUE THREAD

**Issue by hexabinaer** [mautic/mautic#13885]
### Mautic Version

5.0.x series

### Way of installing

Other (please specify if relevant in the bug report)

### PHP version

not relevant

### What browsers are you seeing the problem on?

Not relevant

### What happened?

I was astonished to find that required form elements are solely indicated by a css pseudo element (and a validation feedback). CSS can be a volatile thing. For example, I am currently reviewing a custom template (design-wise). In combination with a WordPress plugin the asterisk attached to the `mauticform-required` wrapper class simply had vanished.

Please note that required form fields should always contain a boolean `required` attribute. 

Given example:
`<input type="email" name="mauticform[email]" value="" id="mauticform_input_email" class="mauticform-input form-control">`

Change to:
`<input type="email" name="mauticform[email]" value="" id="mauticform_input_email" class="mauticform-input form-control" required>`

Quoting from https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/required

> Provide an indication to users informing them the form control is required. Ensure the messaging is multi-faceted, such as through text, color, markings, and attribute, so that all users understand the requirements whether they have color blindness, cognitive differences, or are using a screen reader.

Hoping that this is an easy fix to make life easier for marginalized users (and broaden the potential target groups).

Tags suggestions: ~accessibility ~user-interface

(Funny enough I found that issue re. admin marketers using a screenreader and really wondered if that marketer never tested corresponding outbound forms #10442)

### How can we reproduce this issue?

Look into the source of any given required field ;-)

### Code of Conduct

- [X] I confirm that I have read and agree to follow this project's Code of Conduct

<br /><hr>
Care about this issue? Want to get it resolved sooner? If you are a <a href='https://www.mautic.org/become-a-member-of-mautic'>member of Mautic</a>, you can add some funds to the <a href='https://opencollective.com/mautic/projects/bounties'>Bounties Project</a> so that the person who completes this task can claim those funds once it is merged by a member of the core team! Read the docs <a href='https://contribute.mautic.org/product-team/mautic-bounty-programme'>here.</a>

**Comment by hexabinaer** [comment]
Additionally: according to a colleague using a screen reader the indication "through text" requirement is satisfied by a (real) asterisk next inside the label markup. No need to verbalize "required", screen readers are verbose enough without.

**Comment by kavya-gaur** [comment]
I would like to work on this issue by making it required field.

**Comment by RCheesley** [comment]
Hi there,

We've done an awful lot of work on the UI/UX side of things which was released in the 5.1 release just recently - are you up to date with that yet? cc @mautic/ux-ui-tiger-team for the contribution offers here!

**Comment by andersonjeccel** [comment]
@kavya-gaur Still needed, feel free to go

**Comment by hexabinaer** [comment]
Elaborating andersonjeccel's comment: 
The Mautic I was refering to has been updated in the meantime. A colleague has just checked back with a blank theme and the required attribute is still missing.

**Comment by LordRembo** [comment]
@kavya-gaur did you manage to get started with this issue?

**Comment by andersonjeccel** [comment]
Still relevant
